### PR TITLE
Add cloud-run documentation for testing framework

### PR DESCRIPTION
**Pull request by liamcervante**
Add quick documentation for the new `cloud-run` option introduced in #33710.

### CODE REVIEW

File: website/docs/cli/commands/test.mdx [hunk 1]
```diff
@@ -24,6 +24,8 @@ Terraform then executes a series of Terraform plan or apply commands according t
 
 The following options apply to the Terraform `test` command:
 
+* `-cloud-run=<module source>` - This test run will execute remotely on Terraform Cloud within the specified Terraform [private registry](/terraform/language/modules/sources#terraform-registry) module.
```
Comment (rkoron007) [inline website/docs/cli/commands/test.mdx:27]:
```suggestion
* `-cloud-run=<module source>` - This test run executes remotely on Terraform Cloud within the specified Terraform [private registry](/terraform/language/modules/sources#terraform-registry) module.
```

File: website/docs/cli/commands/test.mdx [hunk 2]
```diff
@@ -43,3 +45,13 @@ The Terraform `test` command creates _real_ infrastructure. Once Terraform fully
 You should monitor the output of the test command closely to ensure Terraform removes the infrastructure it created or perform manual cleanup if not. We recommend creating dedicated testing accounts within the target providers that you can routinely and safely purge to ensure any accidental and costly resources aren't left behind.
 
 Terraform also provides diagnostics explaining why it could now automatically clean up. You should resolve these diagnostics to ensure that future clean-up operations are successful.
+
+## Terraform Cloud Execution
```
Comment (rkoron007) [inline website/docs/cli/commands/test.mdx:49]: sentence case:
```suggestion
## Terraform Cloud execution
```

File: website/docs/cli/commands/test.mdx [hunk 3]
```diff
@@ -43,3 +45,13 @@ The Terraform `test` command creates _real_ infrastructure. Once Terraform fully
...
+
+You can execute your tests remotely on Terraform Cloud using the `-cloud-run` option.
```
Comment (rkoron007) [inline website/docs/cli/commands/test.mdx:51]:
```suggestion
You can execute tests remotely on Terraform Cloud using the `-cloud-run` option.
```

File: website/docs/cli/commands/test.mdx [hunk 4]
```diff
@@ -43,3 +45,13 @@ The Terraform `test` command creates _real_ infrastructure. Once Terraform fully
...
+
+The argument accepts a private registry module source as specified in the [Terraform Registry Module Sources](/terraform/language/modules/sources#terraform-registry) documentation. This will associate the test run with the given private module within the Terraform Cloud UI.
+
+You must provide a module from a **private** registry and not the public Terraform registry.
```
Comment (rkoron007) [inline website/docs/cli/commands/test.mdx:55]: I like the bold, but we use italics for emphasis, so I think we should change to:
```suggestion
You must provide a module from a _private_ registry, not the public Terraform registry.
```

File: website/docs/cli/commands/test.mdx [hunk 5]
```diff
@@ -43,3 +45,13 @@ The Terraform `test` command creates _real_ infrastructure. Once Terraform fully
...
+
+You must execute [`terraform login`](/terraform/cli/command/login) before using this option, and ensure that the provided `hostname` argument matches the private registry hostname of your target module.
```
Comment (rkoron007) [inline website/docs/cli/commands/test.mdx:57]:
```suggestion
You must execute [`terraform login`](/terraform/cli/command/login) before using this option, and ensure that your `hostname` argument matches the private registry hostname of your target module.
```

File: website/docs/cli/commands/test.mdx [hunk 6]
```diff
@@ -43,3 +45,13 @@ The Terraform `test` command creates _real_ infrastructure. Once Terraform fully
 You should monitor the output of the test command closely to ensure Terraform removes the infrastructure it created or perform manual cleanup if not. We recommend creating dedicated testing accounts within the target providers that you can routinely and safely purge to ensure any accidental and costly resources aren't left behind.
 
 Terraform also provides diagnostics explaining why it could now automatically clean up. You should resolve these diagnostics to ensure that future clean-up operations are successful.
+
+## Terraform Cloud Execution
+
+You can execute your tests remotely on Terraform Cloud using the `-cloud-run` option.
+
+The argument accepts a private registry module source as specified in the [Terraform Registry Module Sources](/terraform/language/modules/sources#terraform-registry) documentation. This will associate the test run with the given private module within the Terraform Cloud UI.
```
Comment (rkoron007) [inline website/docs/cli/commands/test.mdx:53]: Can we shorten this to:
```suggestion
The `-cloud-run` command accepts a [private registry module source](/terraform/language/modules/sources#terraform-registry) as as argument. This argument associates the test run with your specified private module within the Terraform Cloud user interface.
```

**Review by rkoron007 — APPROVED** [review]
Tiny suggestions for style, but this looks good to me! ✨

File: website/docs/cli/commands/test.mdx (hunk 6 above)
Comment (liamcervante) [reply website/docs/cli/commands/test.mdx]: Tweaked this a bit more to get around confusino of command vs argument vs option:

```
The `-cloud-run` option accepts a [private registry module source](/terraform/language/modules/sources#terraform-registry). This option associates the test run with your specified private module within the Terraform Cloud user interface.
```
